inference-engine/thirdparty/clDNN/kernel_selector/core/cl_kernels/batch_to_space_ref.cl

```diff
@@ -0,0 +1,82 @@
+{
+    const uint batch = get_global_id(0);
+    const uint feature = get_global_id(1);
+
+#ifdef OUTPUT_LAYOUT_BFYX
+    const uint w = 0;
+    const uint z = 0;
+    const uint y = get_global_id(2) / OUTPUT_SIZE_X;
+    const uint x = get_global_id(2) % OUTPUT_SIZE_X;
+#elif OUTPUT_LAYOUT_BFZYX
+    const uint w = 0;
+    const uint yx = get_global_id(2) % (OUTPUT_SIZE_X * OUTPUT_SIZE_Y);
+    const uint z = get_global_id(2) / (OUTPUT_SIZE_X * OUTPUT_SIZE_Y);
+    const uint y = yx / OUTPUT_SIZE_X;
+    const uint x = yx % OUTPUT_SIZE_X;
+#elif OUTPUT_LAYOUT_BFWZYX
+    const uint zyx = get_global_id(2) % (OUTPUT_SIZE_X * OUTPUT_SIZE_Y * OUTPUT_SIZE_Z);
+    const uint w = get_global_id(2) / (OUTPUT_SIZE_X * OUTPUT_SIZE_Y * OUTPUT_SIZE_Z);
+    const uint yx = zyx % (OUTPUT_SIZE_X * OUTPUT_SIZE_Y);
+    const uint z = zyx / (OUTPUT_SIZE_X * OUTPUT_SIZE_Y);
+    const uint y = yx / OUTPUT_SIZE_X;
+    const uint x = yx % OUTPUT_SIZE_X;
+#endif
+
+    const uint input_feature = (feature + CROPS_BEGIN_FEATURE) / BLOCK_SHAPE_FEATURE;
+    const uint offset_feature = (feature + CROPS_BEGIN_FEATURE) % BLOCK_SHAPE_FEATURE;
+
+    const uint input_w = (w + CROPS_BEGIN_W) / BLOCK_SHAPE_W;
+    const uint offset_w = (w + CROPS_BEGIN_W) % BLOCK_SHAPE_W;
+
+    const uint input_z = (z + CROPS_BEGIN_Z) / BLOCK_SHAPE_Z;
+    const uint offset_z = (z + CROPS_BEGIN_Z) % BLOCK_SHAPE_Z;
+
+    const uint input_y = (y + CROPS_BEGIN_Y) / BLOCK_SHAPE_Y;
+    const uint offset_y = (y + CROPS_BEGIN_Y) % BLOCK_SHAPE_Y;
+
+    const uint input_x = (x + CROPS_BEGIN_X) / BLOCK_SHAPE_X;
+    const uint offset_x = (x + CROPS_BEGIN_X) % BLOCK_SHAPE_X;
+
+    const uint offset_batch = (offset_feature * BLOCK_SHAPE_W * BLOCK_SHAPE_Z * BLOCK_SHAPE_Y * BLOCK_SHAPE_X +
+                               offset_w * BLOCK_SHAPE_Z * BLOCK_SHAPE_Y * BLOCK_SHAPE_X +
+                               offset_z * BLOCK_SHAPE_Y * BLOCK_SHAPE_X +
+                               offset_y * BLOCK_SHAPE_X +
+                               offset_x) * OUTPUT_BATCH_NUM;
+    const uint input_batch = batch + offset_batch;
+
+    const uint input_index = INPUT0_OFFSET +
+                             input_batch * INPUT0_BATCH_PITCH +
+                             input_feature * INPUT0_FEATURE_PITCH +
+                             input_w * INPUT0_W_PITCH +
+                             input_z * INPUT0_Z_PITCH +
+                             input_y * INPUT0_Y_PITCH +
+                             input_x;
+
+    const uint output_index = OUTPUT_OFFSET +
+                              batch * OUTPUT_BATCH_PITCH +
+                              feature * OUTPUT_FEATURE_PITCH +
+                              w * OUTPUT_W_PITCH +
+                              z * OUTPUT_Z_PITCH +
+                              y * OUTPUT_Y_PITCH +
+                              x;
+
+    output[output_index] = ACTIVATION(input[input_index], ACTIVATION_PARAMS);
```

Comment:
`ACTIVATION `macro do nothing until you add your primitive into "White list" at `void prepare_primitive_fusing::fuse_activations(program_impl &p)` function in `prepare_primitive_fusing.cpp`